### PR TITLE
Add Image Resizing

### PR DESCRIPTION
**Pull request by thejsj**

### CODE REVIEW

File: basilica/basilica/__init__.py [hunk 1]
```diff
@@ -126,6 +128,9 @@ def embed_images(self, images, model='generic', version='default',
         """
         url = '%s/embed/images/%s/%s' % (self.server, model, version)
         data = ({'img': base64.b64encode(img).decode('utf-8')} for img in images)
+        if (type(images) == list):
```
Comment (mlucy) [inline basilica/basilica/__init__.py]: This should work for all enumerables, not just lists.

File: basilica/basilica/__init__.py [hunk 2]
```diff
@@ -126,6 +128,9 @@ def embed_images(self, images, model='generic', version='default',
...
+            images = map(self.__resize_image, images)
+            data = ({'img': base64.b64encode(img).decode('utf-8')} for img in images)
```
Comment (mlucy) [inline basilica/basilica/__init__.py]: This is also being done on line 130.  I think we should delete lines 130-133 and write:
```
data = ({'img': base64.b64encode(self.__resize_image(img)).decode('utf-8')} for img in images)
```

File: basilica/basilica/__init__.py [hunk 3]
```diff
@@ -197,7 +202,7 @@ def embed_image_files(self, image_files, model='generic', version='default',
         def load_image_files(image_files):
             for image_file in image_files:
                 with open(image_file, 'rb') as f:
-                    yield f.read()
+                    yield self.__resize_image(f.read())
```
Comment (mlucy) [inline basilica/basilica/__init__.py]: I don't think we need to do this here, it's already being done in `embed_images`.

File: basilica/basilica/__init__.py [hunk 4]
```diff
@@ -303,3 +308,14 @@ def embed_sentence(self, sentence, model='english', version='default',
         """
         return list(self.embed_sentences([sentence], model=model, version=version,
                                          opts=opts, timeout=timeout))[0]
+
+    def __resize_image(self, image):
+        if type(image) != bytes:
+            raise ValueError('`image` argument must be bytes (got `%s`)' % (type(image)))
+        im = Image.open(BytesIO(image))
+        im.thumbnail((256, 256))
+        im = im.convert("RGB")
```
Comment (mlucy) [inline basilica/basilica/__init__.py]: Why do we need to do this RGB conversion?

File: basilica/basilica/__init__.py [hunk 5]
```diff
@@ -303,3 +308,14 @@ def embed_sentence(self, sentence, model='english', version='default',
         """
         return list(self.embed_sentences([sentence], model=model, version=version,
                                          opts=opts, timeout=timeout))[0]
+
+    def __resize_image(self, image):
+        if type(image) != bytes:
+            raise ValueError('`image` argument must be bytes (got `%s`)' % (type(image)))
+        im = Image.open(BytesIO(image))
```
Comment (mlucy) [inline basilica/basilica/__init__.py]: What's the error behavior if someone tries to open an unsupported image type?

**Review by mlucy — CHANGES_REQUESTED** [review]
Some toplevel comments:

* After some though, I think we should actually resize to 512x512.  The network costs aren't that much larger, and if we deploy a newer image model it will probably want 512x512 input since that's becoming more common, and it would be nice if people didn't need to update their clients.
* I think there should be an option to turn off the automatic image resizing, just so that it's user-controllable in some future case we aren't considering.  (We could probably skimp on documenting that for now.)
* The version needs to be bumped.  This requires editing `setup.py` and `README.rst`.

**Comment by mlucy** [comment]
This looks good!  It's awesome that it's so straightforward.

**Comment by mlucy** [comment]
Also, we should probably add some tests so that we don't accidentally break this.  (We currently don't have unit tests for the client, which is maybe a mistake, so we'd have to add them to the basilica-django tests :/.)

File: basilica/basilica/__init__.py (hunk 4 above)
Comment (thejsj) [reply basilica/basilica/__init__.py]: If it's a PNG, it's rbga.

**Pull request merged**